### ISSUE TITLE
Event Registration Templates - When the 'Group Leaders' notification checkbox is checked, all notifications fail to send.

### ISSUE THREAD

**Issue by tstephens17** [SparkDevNetwork/Rock#3127]
### Prerequisites
* [X] Put an X between the brackets on this line if you have done all of the following:
  * Can you reproduce the problem on a fresh install or the [demo site ](http://rock.rocksolidchurchdemo.com/)?
  * Did you include your Rock version number and [client culture ](https://github.com/SparkDevNetwork/Rock/wiki/Environment-and-Diagnostics-Information) setting?
  * Did you [?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ASparkDevNetwork%2FSlack perform a cursory search ](https://github.com/issues) to see if your bug or enhancement is already reported?

### Description
We updated from 6.10 to 7.4 and started experiencing issues with Event Registration notification emails to registration contacts (Not all but some). We asked on Slack and no one else was experiencing issues. They even said everything was working okay. So, I began to look deeper, trying different settings, making sure the [Notify] field in the database was being set. This is when I discovered that if the "Group Leaders" option was selected, suddenly the notification email to the contact stopped being sent. Simply unchecking this option allowed them to start working again. In this case we did not have a group leader or a group set. So, unsure if the issue is just the setting or not having a group.

### Steps to Reproduce
1. Create an registration template and check all three boxes for the notify settings.
1. Set an event contact
1. Complete the needed steps in order to be able to register for the event.
1. Notice that the event contact is not emailed.
1. Uncheck "group leaders" in the notify settings on the template.
1. Sign up again for the event.
1. Notice that the contact is not emailed when someone is registered.

**Expected behavior:**

I expect the event contact to be emailed if the box is checked.

**Actual behavior:**

The check box is ignored if "Group Leaders" is checked.

### Versions
* **Rock Version:** [Rock McKinley 7.4 (1.7.4.58)]
* **Client Culture Setting:** [en-US]

**Comment by mikejed** [comment]
![image](https://user-images.githubusercontent.com/6932047/42651648-db1bb290-85c4-11e8-9c24-0087d5680a02.png)

**Comment by bobrufenacht** [comment]
I have experienced this issue when Registration Contact and Group Leaders are selected in the template, but there is no group specified on the Event Occurrence Registration Instance.  If Group Leaders (or Group Followers) are selected on the Template and there is no target group, nobody including the Registration Contact is notified.  If I added a group (even with no members), My registration contact got the email.  If I unchecked the Group Leader on the template then the Registration Contact was notified as well.  So adding a group or removing the checkbox was my work-around for now.
